### ISSUE TITLE
Code cells with continuation lines are not supported.

### ISSUE THREAD

**Issue by siko1056** [gnu-octave/pkg-jupyter-notebook#1]
Greetings @Abdallah-Elshamy,

In one of my notebooks, I stumbled over this issue.  Given this code cell:

```octave
p = [ ...
  50 - 30i; ...
  18 +  8i; ...
  12 - 10i; ...
 -14 - 60i; ...
  40 + 20i];
```

That [line continuations](https://octave.org/doc/v6.3.0/Continuation-Lines.html) are not supported.

This is probably due to the artificial splitting of each line.

https://github.com/gnu-octave/pkg-jupyter-notebook/blob/2f96cad2949cca35eb004a38bb22b275e9ba5e49/src/JupyterNotebook.m#L162-L179

Is there a reason, why this has to be done and the cell is not evaluated as a whole?  In the end they seem to get merged anyway.

**Comment by Abdallah-Elshamy** [comment]
> Is there a reason, why this has to be done and the cell is not evaluated as a whole? In the end they seem to get merged anyway.

There will be a problem with errors. In Jupyter notebooks, the execution will continue if there is an error:
![Screenshot from 2021-07-25 10-11-26](https://user-images.githubusercontent.com/33662044/126892412-b81097cc-8404-43c8-b445-93657bcbf943.png)

trying the same code using `evalc`:
```
k = strjoin(c.notebook.cells{1}.source);
evalc(k, "printf (\"error: \"); printf (lasterror.message)")
```
will result in:
```
ans = ans = qt
error: 'd' undefined near line 1 column 2
```
without the rest of the code being executed.

**Comment by siko1056** [comment]
Thanks for the example :+1:   I find it indeed a little weird how the Jupyter Octave kernel "fixes" the execution.  

Personally, and for the sake of simplicity, I prefer valid Octave code execution and would feed all input at once to evalc.  An error is an error.  Most likely the continuous script execution is worthless anyway, when the programmer made a mistake.

However, if you are keen to simulate the Jupyter Kernel program, we should probably enable both behaviors.

As this item seems to be more complicated, let us **defer it to the end of GSoC** and get opinions from the community if they prefer all Octave code to be run or a continued script execution in the event of an error :slightly_smiling_face:

**Comment by Abdallah-Elshamy** [comment]
> Personally, and for the sake of simplicity, I prefer valid Octave code execution and would feed all input at once to evalc. An error is an error. Most likely the continuous script execution is worthless anyway, when the programmer made a mistake.

I agree with that. I definitely prefer to have line continuations and basic strucures (like loops and functions)  than resuming executing after errors. Besides, it is better in terms of performance as we don't need to load and store the context for every line in a cell.   

> However, if you are keen to simulate the Jupyter Kernel program, we should probably enable both behaviors. That is either:
> 
>     * looking for "..."
> 
>     * run evalc in try-catch blocks and continue the execution from the erroneous line +1
> 
>     * ...?
> 
> 
> Both looks very ugly to me sweat
> 

I was thinking about that too and I couldn't find a good solution for it.

> As this item seems to be more complicated, let us **defer it to the end of GSoC** and get opinions from the community if they prefer all Octave code to be run or a continued script execution in the event of an error slightly_smiling_face

I will change it to the whole cell for now and if the majority wants the continued script execution (which I highly doubt since loops and conditions are affected too), we can easily return it.

**Comment by Abdallah-Elshamy** [comment]
Fixed in 7faf1da7244eadeb46fb0c0db1c3e294a2ab5a1d and 006d4a45c889641af9b67b1d7c147040e5f5736c
